src/diffpy/utils/diffraction_objects.py

```diff
@@ -169,32 +175,55 @@ def __eq__(self, other):
-            raise TypeError("I only know how to sum two DiffractionObject objects")
-        elif self.on_tth[0].all() != other.on_tth[0].all():
-            raise RuntimeError(x_grid_emsg)
-        else:
-            summed.on_tth[1] = self.on_tth[1] + other.on_tth[1]
-            summed.on_q[1] = self.on_q[1] + other.on_q[1]
-        return summed
+        """Add a scalar value or another DiffractionObject to the xarrays of
+        the DiffractionObject.
 
-    def __radd__(self, other):
-        summed = deepcopy(self)
-        if isinstance(other, int) or isinstance(other, float) or isinstance(other, np.ndarray):
-            summed.on_tth[1] = self.on_tth[1] + other
-            summed.on_q[1] = self.on_q[1] + other
-        elif not isinstance(other, DiffractionObject):
-            raise TypeError("I only know how to sum two Scattering_object objects")
-        elif self.on_tth[0].all() != other.on_tth[0].all():
-            raise RuntimeError(x_grid_emsg)
+        Parameters
+        ----------
+        other : DiffractionObject or int or float
+            The object to add to the current DiffractionObject. If `other` is a scalar value,
+            it will be added to all xarrays. The length of the xarrays must match if `other` is
+            an instance of DiffractionObject.
+
+        Returns
+        -------
+        DiffractionObject
+            The new and deep-copied DiffractionObject instance after adding values to the xarrays.
+
+        Raises
+        ------
+        ValueError
+            Raised when the length of the xarrays of the two DiffractionObject instances do not match.
+        TypeError
+            Raised when the type of `other` is not an instance of DiffractionObject, int, or float.
+
+        Examples
+        --------
+        Add a scalar value to the xarrays of the DiffractionObject instance:
+        >>> new_do = my_do + 10.1
+
+        Add the xarrays of two DiffractionObject instances:
+        >>> new_do = my_do_1 + my_do_2
+        """
+
+        summed_do = deepcopy(self)
+        # Add scalar value to all xarrays by broadcasting
+        if isinstance(other, (int, float)):
+            summed_do._all_arrays[:, 1] += other
+            summed_do._all_arrays[:, 2] += other
+            summed_do._all_arrays[:, 3] += other
+        # Add xarrays of two DiffractionObject instances
+        elif isinstance(other, DiffractionObject):
+            if len(self.on_tth()[0]) != len(other.on_tth()[0]):
+                raise ValueError(x_grid_length_mismatch_emsg)
+            summed_do._all_arrays[:, 1] += other.on_q()[0]
+            summed_do._all_arrays[:, 2] += other.on_tth()[0]
+            summed_do._all_arrays[:, 3] += other.on_d()[0]
         else:
-            summed.on_tth[1] = self.on_tth[1] + other.on_tth[1]
-            summed.on_q[1] = self.on_q[1] + other.on_q[1]
-        return summed
+            raise TypeError(invalid_add_type_emsg)
+        return summed_do
+
+    __radd__ = __add__
```

Comment:
__radd__ is back - used stackoverflow: https://stackoverflow.com/questions/9126766/addition-between-classes-using-radd-method